### PR TITLE
Data jpa

### PR DESCRIPTION
**Pull request by Paradoxallist**

### CODE REVIEW

File: src/main/java/store/bookshop/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -37,4 +37,16 @@ public BookDto findById(Long id) {
                 () -> new EntityNotFoundException("Can't find book by id: " + id));
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto updateById(CreateBookRequestDto requestDto, Long id) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
```
Comment (Rommelua) [inline src/main/java/store/bookshop/service/impl/BookServiceImpl.java:49]: check if id exists first

Comment (koval-kn) [reply src/main/java/store/bookshop/service/impl/BookServiceImpl.java:49]: not fixed. you can check if id exists by getting book by id from db

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/store/bookshop/model/Book.java [hunk 2]
```diff
@@ -5,34 +5,37 @@
...
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Getter
 @Setter
 @NoArgsConstructor
-@RequiredArgsConstructor
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/store/bookshop/model/Book.java]:
```suggestion
@SQLRestriction(value = "is_deleted = FALSE")
```

File: src/main/java/store/bookshop/model/Book.java [hunk 3]
```diff
@@ -5,34 +5,37 @@
...
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Getter
 @Setter
 @NoArgsConstructor
-@RequiredArgsConstructor
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
```
Comment (okuzan) [inline src/main/java/store/bookshop/model/Book.java]:
```suggestion
@SQLDelete(sql = "UPDATE books SET is_deleted = TRUE WHERE id = ?")
```

File: src/main/java/store/bookshop/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -37,4 +37,19 @@ public BookDto findById(Long id) {
                 () -> new EntityNotFoundException("Can't find book by id: " + id));
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto updateById(CreateBookRequestDto requestDto, Long id) {
+        if (bookRepository.findById(id).isEmpty()) {
+            throw new RuntimeException("There is no exist book with this id: " + id);
```
Comment (okuzan) [inline src/main/java/store/bookshop/service/impl/BookServiceImpl.java]: better throw more specific entitynotfound, as in your other methods

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-book-table.yaml
+  - include:
+      file: db/changelog/changes/02-insert-books.yaml
```
Comment (okuzan) [inline src/main/resources/db/changelog/db.changelog-master.yaml]:
```suggestion
      file: db/changelog/changes/02-insert-books.yaml

```
forgot an empty line

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/store/bookshop/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -37,4 +37,19 @@ public BookDto findById(Long id) {
                 () -> new EntityNotFoundException("Can't find book by id: " + id));
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto updateById(CreateBookRequestDto requestDto, Long id) {
+        if (bookRepository.findById(id).isEmpty()) {
+            throw new EntityNotFoundException("There is no exist book with this id: " + id);
```
Comment (MAMentorFX) [inline src/main/java/store/bookshop/service/impl/BookServiceImpl.java]:
```suggestion
            throw new EntityNotFoundException("There is no existing book with this id: " + id);
```
or better
```suggestion
            throw new EntityNotFoundException("Book doesn't exist with id: " + id);
```

File: src/main/java/store/bookshop/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -37,4 +37,19 @@ public BookDto findById(Long id) {
...
+        }
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
```
Comment (MAMentorFX) [inline src/main/java/store/bookshop/service/impl/BookServiceImpl.java]: Make Transactional

File: src/main/java/store/bookshop/service/impl/BookServiceImpl.java (hunk 1 above)
Comment (MAMentorFX) [reply src/main/java/store/bookshop/service/impl/BookServiceImpl.java:49]: fixed

**Review by MAMentorFX — APPROVED** [review]

**Pull request merged**